### DISCUSSION TITLE
Search with Market Context

### DISCUSSION THREAD

**Discussion by mikkopaderes-iona** [Shopify/storefront-api-feedback#208] [Help]
I'm trying to search in the context of a market. I tried using `@inContext` with country and language but it doesn't work. The search always seems to return every product exposed to the sales channel (Hydrogen or Shopify GraphiQL App in my case). Here's my query:

```graphql
query SearchProducts($country: CountryCode, $language: LanguageCode) @inContext(country: $country, language: $language) {
  search(
    query: "2020"
    types: PRODUCT
    sortKey: RELEVANCE
    unavailableProducts: LAST
    first: 5
  ) {
    totalCount
    pageInfo {
      endCursor
      hasNextPage
    }
    nodes {
      ... on Product {
        handle
        title
      }
    }
  }
}
```

Is Market context not supported for search? Because under the `collection` query, it does work and only shows the product depending on what I set in `@inContext`.

**Comment by johncraigcole** [comment]
Hey @mikkopaderes-iona thanks for sharing this with the community. Verified the issue with my own testing and an issue has been created to get this looked at and resolved. I don't have a timeline on when resolution will happen, but will follow up here when it does.

**Reply by johncraigcole** [reply]
Hey @mikkopaderes-iona a PR just landed in production to fix this issue. If the product is excluded from a market, both a search or product query will no longer return that product.

**Reply by mikkopaderes-iona** [reply]
Thank you! We went for a metafield approach that stores what market they are published in while waiting for this. But we'll definitely try this fix out.
